packages/peregrine/lib/talons/SearchBar/useAutocomplete.js

```diff
@@ -11,43 +11,48 @@ import debounce from 'lodash.debounce';
  * Returns props necessary to render an Autocomplete component.
  * @param {Object} props
  * @param {DocumentNode} props.query - GraphQL query
- * @param {Boolean} props.visible - whether to show
+ * @param {Boolean} props.valid - whether to run the query
+ * @param {Boolean} props.visible - whether to show the element
  */
 export const useAutocomplete = props => {
-    const { query, visible } = props;
+    const { query, valid, visible } = props;
 
     // prepare to run query
     const [runQuery, queryResult] = useLazyQuery(query);
     const { data, error, loading } = queryResult;
 
     // retrieve value from another field
     const { value } = useFieldState('search_query');
-    const valid = value && value.length > 2;
 
     // Create a debounced function so we only search some delay after the last
     // keypress.
-    const debouncedRunQuery = useMemo(() => debounce(runQuery, 500), [
-        runQuery
-    ]);
+    const debouncedRunQuery = useMemo(
+        () =>
+            debounce(inputText => {
+                runQuery({ variables: { inputText } });
```

Comment:
This now runs with the last 3 characters in the input when you start deleting some text value. For example "top shirt" will search "top shirt" but when you delete that character by character it'll query for "top" since it is the last valid value on the way to an empty string.

I would prefer we don't make an extra call for text that may never be searched but it's also a minor optimization so :shrug: